src/main/java/com/arpnetworking/metrics/impl/TsdMetrics.java

```diff
@@ -109,14 +110,14 @@ public void incrementCounter(final String name, final long value) {
         }
 
         final ConcurrentLinkedDeque<Quantity> counters = getOrCreateDeque(_counterSamples, name);
-        TsdCounter counter = _counters.get(name);
-        if (counter == null) {
-            final TsdCounter newCounter = TsdCounter.newInstance(name, _isOpen);
-            counter = getOrCreate(_counters, name, newCounter);
-            if (counter == newCounter) {
-                counters.add(counter);
+        final TsdCounter counter = _counters.compute(name, (key, existingValue) -> {
```

Comment:
Wait a darn minute. Brandon is having me _remove_ a lambda? ;)

This lambda uses a captures over counters so removing it was a little trickier. Take a look.